### PR TITLE
Decompiled En_Holl

### PR DESCRIPTION
**Pull request by Thar0**
I don't like submitting nonmatching actors but my hand has been forced.
1 non-matching remaining, regalloc only.
Partially documented.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c [hunk 1]
```diff
@@ -21,33 +36,355 @@ const ActorInit En_Holl_InitVars = {
...
+            setupDLIdx = 0;
+        }
+        gfxP = Gfx_CallSetupDL(gfxP, setupDLIdx);
+        if (this->side == 0) {
+            Matrix_RotateY(M_PI, MTXMODE_APPLY);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Holl/EnHoll_Update.s")
+        gSPMatrix(gfxP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_holl.c", 824), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gDPSetPrimColor(gfxP++, 0, 0, 0, 0, 0, (u8)this->planeAlpha);
```
[the remark this answers is not in the document]
Comment (Thar0) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c:376]: Cast is necessary to match

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c [hunk 2]
```diff
@@ -21,33 +36,355 @@ const ActorInit En_Holl_InitVars = {
...
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP)
+};
+
+/**
+ * These are all distances in the relative z direction. That is, moving towards or away
+ *   from the "face" of the loading plane regardless of orientation.
+ * [0] : Load the room on this side of the loading plane if not already loaded
+ * [1] : Load the room on the other side of the loading plane
```
[the remark this answers is not in the document]
Comment (Thar0) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c:55]: I updated the comment to try and clarify a bit more, would appreciate feedback on the change

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c [hunk 3]
```diff
@@ -21,33 +36,355 @@ const ActorInit En_Holl_InitVars = {
...
+ *   Load Current Room (fails as it is already loaded)
+ *   Load Next Room
+ *   Load Previous Room
+ *   Load Next Room
+ * 
+ *  @bug The striped nature of loading planes can cause some actors to unload due to
+ *      conflicting Object Lists between the two rooms
+ * 
+ *  @bug If you can get around to the other side of the loading plane without triggering it,
+ *      you can load the room on the other side multiple times
```
[the remark this answers is not in the document]
Comment (Thar0) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c]: Do you have any suggestions as to where these comments could be moved to? I feel like there's no one place in the code that contributes to these bugs so wasn't sure where to put it in the code itself, this was more of a temporary place just to have those documented.

[the remark this answers is not in the document]
Comment (Roman971) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c]: I'm not sure, but I think somewhere more general like init or update would already be better.
Either way, I'm also not convinced the fact that actors can unload between rooms belongs as a bug, or at least it's not limited to this actor's behaviour.

[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c]: I said the same thing. It feels like the bug is moreso how you can get around to the other side of these in the first place.
But tharo said that ideally the bug is fixed in this actor (for future modding purposes), so it should definitely be mentioned somewhere in this file, but maybe not with a bug tag

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (Roman971) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c:55]: I'm not sure this is the best way to describe them, but with this new sentence the comments on the last 2 values should at least be updated.

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (Roman971) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c]: Well I was referring specifically about the bug on "Object Lists conflicts", which is a general issue with how room transitions are handled. Tho it feels more like a limitation than a bug to me.

[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c]: oh I see. the problem is specific to this implementation of different "stripes" of loading triggers for the black planes though, right? obv the root cause is object dependencies in general, but it really becomes an issue for this specific method of loading the next room

[the remark this answers is not in the document]
Comment (Roman971) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c]: I don't think so, but I could be wrong.

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (fig02) [reply src/overlays/actors/ovl_En_Holl/z_en_holl.c:55]: if we can take some time in discord soon to discuss these remaining documentation issues itd be good i think

**Comment by fig02** [comment]
i forget is there anything we need to figure out with this file?

**Comment by fig02** [comment]
do we want to merge and get back to the documentation stuff at a later time? I think its probably the best move.
Just got some conflicts to resolve @Thar0

**Review by fig02 — APPROVED** [review]
code looks good to me, and like we said we'll get to docs another time :D

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c [hunk 4]
```diff
@@ -21,33 +36,351 @@ const ActorInit En_Holl_InitVars = {
...
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Holl/func_80A59618.s")
+void EnHoll_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnHoll* this = THIS;
+    Gfx* gfxP;
+    u32 setupDLIdx;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Holl/z_en_holl.c:356]: This function needs to be updated to use OPEN_DISPS/CLOSE_DISPS.

**Review by Roman971 — CHANGES_REQUESTED** [review]

**Review by Roman971 — APPROVED** [review]

**Pull request merged**